File: baekjoon/단계별로 풀어보기/집합과 맵/1764_듣보잡.py
# https://www.acmicpc.net/problem/1764

from sys import stdin

input = stdin.readline
a, b = map(int, input().split())
A = set()
B = set()
for _ in range(a):
    A.add(input().rstrip())
for _ in range(b):
    B.add(input().rstrip())
u = sorted(list(A.intersection(B)))
print(len(u))
for x in u:
    print(x)

File: C#/Baekjoon/집합과 맵/1764_듣보잡.cs
//https://www.acmicpc.net/problem/1764

using var sw = new StreamWriter(new BufferedStream(Console.OpenStandardOutput()));
int[] n = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
var A = new HashSet<string>();
var B = new HashSet<string>();
for (int i = 0; i < n[0]; i++) A.Add(Console.ReadLine());
for (int j = 0; j < n[1]; j++) B.Add(Console.ReadLine());
A.IntersectWith(B);
string[] s = A.ToArray();
Array.Sort(s);
sw.WriteLine(s.Length);
foreach (string s2 in s) sw.WriteLine(s2);